cloudformation/iam_federation.js

```diff
@@ -92,15 +105,16 @@ function removeStatement(input, callback) {
   console.log(assumeDoc.Statement);
   console.log(input.roleArn);
   var found = -1;
-  for (var i = 0; i < assumeDoc.Statement.length; i++) {
```

Comment:
Didnt get your point, since the changes makes sure that while adding the new statements they are added to an array while they are removed they would be removed from the same array.